cpp/include/cudf/replace.hpp

```diff
@@ -16,11 +16,63 @@
 
 #pragma once
 
-#include "cudf/cudf.h"
-#include "cudf/types.hpp"
-#include "cudf/column/column.hpp"
+#include <cudf/types.hpp>
+#include <memory>
 
 namespace cudf {
+namespace experimental {
+  /**
+   * @brief Replaces all null values in a column with corresponding values of another column
+   *
```

Comment:
Add something like this:
```suggestion
   *
   * If `input[i]` is NULL, then `output[i]` will contain `replacement[i]`. 
```